addons/l10n_latam_invoice_document/wizards/account_debit_note.py

```diff
@@ -5,11 +5,16 @@ class AccountDebitNote(models.TransientModel):
 
     _inherit = 'account.debit.note'
 
-    def create_debit(self):
-        """ Properly compute the latam document type of type debit note. """
-        res = super().create_debit()
-        new_move_id = res.get('res_id')
-        if new_move_id:
-            new_move = self.env['account.move'].browse(new_move_id)
-            new_move._compute_l10n_latam_document_type()
-        return res
+    def _prepare_default_values(self, move):
+        default_values = super()._prepare_default_values(move)
+
+        # properly compute debit type
+        debit_note = self.env['account.move'].new({
+            'move_type': default_values.get('move_type'),
+            'journal_id': default_values.get('journal_id'),
+            'partner_id': move.partner_id.id,
+            'company_id': move.company_id.id,
+        })
+        document_types = debit_note.l10n_latam_available_document_type_ids.filtered(lambda x: x.internal_type == 'debit_note')
+        default_values['l10n_latam_document_type_id'] = document_types and document_types[0].ids[0]
```

Comment:
```suggestion
        default_values = super()._prepare_default_values(move)

        default_values['l10n_latam_document_type_id'] = self.env['account.move'].browse(default_values['debit_origin_id']).l10n_latam_document_type_id.id
```
Maybe we can do something like this instead? This way we make sure that we get the same document type that was used before